src/PatchOrchestrationApplication/NodeAgentSFUtility/src/Helpers/HealthManagerHelper.cs

```diff
@@ -49,23 +54,92 @@ class HealthManagerHelper
 
             try
             {
-                ServiceHealthReport healthReport = new ServiceHealthReport(new Uri(applicationName + ServiceNameSuffix), healthInformation);
-                fabricClient.HealthManager.ReportHealth(healthReport);
+                ServiceHealthReport healthReport = new ServiceHealthReport(new Uri(applicationName + serviceNameSuffix), healthInformation);
+                Func<string, bool> condition = (s) => {
+                    bool serviceExists = CheckIfServiceIsUp(fabricClient, applicationName, s).GetAwaiter().GetResult();
+                    if (serviceExists)
+                    {
+                        return true;
+                    }
+                    else
+                    {
+                        return false;
+                    }
```

Comment:
you dont need the serviceExists block and instead return CheckIfServiceIsUp(...)

Reply:
thanks